firmware/application/apps/ui_looking_glass_app.cpp

```diff
@@ -519,70 +520,42 @@ GlassView::GlassView(
 
 void GlassView::load_Presets() {
     File presets_file;
-    auto result = presets_file.open("LOOKINGGLASS/PRESETS.TXT");
-    presets_db.clear();  // Start with fresh db
-    if (result.is_valid()) {
-        presets_Default();  // There is no txt, store a default range
-    } else {
-        std::string line;  // There is a txt file
-        char one_char[1];  // Read it char by char
-        for (size_t pointer = 0; pointer < presets_file.size(); pointer++) {
-            presets_file.seek(pointer);
-            presets_file.read(one_char, 1);
-            if ((int)one_char[0] > 31) {       // ascii space upwards
-                line += one_char[0];           // Add it to the textline
-            } else if (one_char[0] == '\n') {  // New Line
-                txtline_process(line);         // make sense of this textline
-                line.clear();                  // Ready for next textline
-            }
+    auto error = presets_file.open("LOOKINGGLASS/PRESETS.TXT");
+    presets_db.clear();
+
+    if (!error) {
+        auto reader = FileLineReader(presets_file);
+        for (const auto& line : reader) {
+            if (line.length() == 0 || line[0] == '#')
+                continue;
+
+            auto cols = split_string(line, ',');
+            if (cols.size() != 3)
+                continue;
+
+            // TODO: add some conversion helpers that take string_view.
+            presets_db.emplace_back(preset_entry{
+                std::stoi(std::string{cols[0]}),
+                std::stoi(std::string{cols[1]}),
+                std::string{cols[2]}});
```

Comment:
trim to be sure the /r/n is gone

Reply:
And will skip anything that does not look like a valid entry, which is good 👍